### PR TITLE
Bug fixes 2

### PR DESCRIPTION
**Pull request by eriese**
fixes are noted in commit messages

### CODE REVIEW

File: app/javascript/components/widgets/encounterCalendar/EncounterCalendar.vue [hunk 1]
```diff
@@ -66,7 +66,10 @@
 
 				for (let i = 0; i < this.selectedPartners.length; i++) {
 					let partner = this.selectedPartners[i];
-					// let partner = this.partnerships[partnerID];
+					if (partner.encounters === undefined) {
+						continue;
+					}
+
```
[the remark this answers is not in the document]
Comment (eriese) [reply app/javascript/components/widgets/encounterCalendar/EncounterCalendar.vue]: Won't this result in looping twice? I'll use `forEach`, but I'm wary of using convenience methods that save me keystrokes but use more steps

[the remark this answers is not in the document]
Comment (novacourtois) [reply app/javascript/components/widgets/encounterCalendar/EncounterCalendar.vue]: yea, i guess you're right about that point.
no real need to filter if you're about to loop through it.

File: app/javascript/components/widgets/EncounterCalendar.vue [hunk 2]
```diff
@@ -120,6 +118,7 @@
 					}
 				}
 
+				max = max || new Date();
```
Comment (novacourtois) [inline app/javascript/components/widgets/EncounterCalendar.vue]: good catch with checking if max was null here

File: app/javascript/components/widgets/EncounterCalendar.vue [hunk 3]
```diff
@@ -1,13 +1,13 @@
 <template>
 	<div id="encounter-calendar">
-		<div v-if="partnerships.length > 1">
+		<div v-if="partnerships.length > 1 && selectedEncounters.length">
 			<v-select v-model="selectedPartners" multiple :options="availablePartners" label="display" :close-on-select="false" :no-drop="empty" :searchable="!empty">
 				<template v-slot:selected-option="opt">
 					<span><span :class="`partnership-${opt.index}`" class="partner-indicator"></span>{{opt.display}}</span>
 				</template>
 			</v-select>
 		</div>
```
Comment (novacourtois) [inline app/javascript/components/widgets/EncounterCalendar.vue]: i found what im calling a bug and i think this is where it's occurring:

from the select tags, if i have one partner selected, and deselect this partner, i don't expect the select to disappear. from a ui perspective, i think it's just as reasonable to assume a user may deselect a partner before selecting a partner, but this doesn't allow for this:
https://gyazo.com/f65cc888f460bfa3547949bd6cf5d1cc

Comment (eriese) [reply app/javascript/components/widgets/EncounterCalendar.vue]: yep, this is a bug I introduced. Thanks for catching it. I'll fix tomorrow

**Pull request merged**